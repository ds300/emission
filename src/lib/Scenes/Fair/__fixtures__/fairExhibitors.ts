export const fairFixture = {
  id: "art-palm-springs-2019",
  _id: "5c17b160ffbbc72eef33040c",
  exhibitors_grouped_by_name: [
    {
      letter: "A",
      exhibitors: [
        {
          name: "Adamar Fine Arts",
          id: "adamar-fine-arts",
          profile_id: "adamar-fine-arts-adamar-fine-arts-at-art-palm-springs-2019",
          partner_id: "54b40a37776f720e459d0500",
        },
        {
          name: "ARCADIA CONTEMPORARY",
          id: "arcadia-contemporary",
          profile_id: "arcadia-contemporary-arcadia-contemporary-at-art-palm-springs-2019",
          partner_id: "53272a99139b2113c10007b6",
        },
        {
          name: "Artmix",
          id: "artmix",
          profile_id: "artmix-artmix-at-art-palm-springs-2019",
          partner_id: "5c2e50530cf1100007b437f7",
        },
        {
          name: "Artspace Warehouse",
          id: "artspace-warehouse",
          profile_id: "artspace-warehouse-artspace-warehouse-at-art-palm-springs-2019",
          partner_id: "5258782e8b3b817b36000e7f",
        },
        {
          name: "Axiom Contemporary",
          id: "axiom-contemporary",
          profile_id: "axiom-contemporary-axiom-contemporary-at-art-palm-springs-2019-1",
          partner_id: "54b40a7e776f720e45da0500",
        },
      ],
    },
    {
      letter: "B",
      exhibitors: [
        {
          name: "Blue Rain Gallery",
          id: "blue-rain-gallery",
          profile_id: "blue-rain-gallery-blue-rain-gallery-at-art-palm-springs-2019",
          partner_id: "58b4acca139b2178c2554ab6",
        },
        {
          name: "Boulderton Contemporary",
          id: "boulderton-contemporary",
          profile_id: "boulderton-contemporary-boulderton-contemporary-at-art-palm-springs-2019",
          partner_id: "5c2e50540cf1100007b437fa",
        },
        {
          name: "Bruce Lurie Gallery",
          id: "bruce-lurie-gallery",
          profile_id: "bruce-lurie-gallery-bruce-lurie-gallery-at-art-palm-springs-2019",
          partner_id: "54b40aca776f720e45020600",
        },
      ],
    },
    {
      letter: "C",
      exhibitors: [
        {
          name: "C. Grimaldis Gallery",
          id: "c-grimaldis-gallery",
          profile_id: "c-grimaldis-gallery-c-grimaldis-gallery-at-art-palm-springs-2019",
          partner_id: "524b45c08b3b81220500024e",
        },
        {
          name: "Caldwell Snyder Gallery",
          id: "caldwell-snyder-gallery",
          profile_id: "caldwell-snyder-gallery-caldwell-snyder-gallery-at-art-palm-springs-2019",
          partner_id: "55ca4146726169709c000147",
        },
        {
          name: "Caviar20",
          id: "caviar20",
          profile_id: "caviar20-caviar20-at-art-palm-springs-2019",
          partner_id: "5227681aebad644079000137",
        },
      ],
    },
    {
      letter: "D",
      exhibitors: [
        {
          name: "Donna Schneier Fine Arts",
          id: "donna-schneier-fine-arts",
          profile_id: "donna-schneier-fine-arts-donna-schneier-fine-arts-at-art-palm-springs-2019",
          partner_id: "59c2a379ab60ca1153bad492",
        },
      ],
    },
    {
      letter: "F",
      exhibitors: [
        {
          name: "Fabrik Projects Gallery",
          id: "fabrik-projects-gallery",
          profile_id: "fabrik-projects-gallery-fabrik-projects-gallery-at-art-palm-springs-2019",
          partner_id: "5941712db202a34d5c754b35",
        },
        {
          name: "Fourth Element Gallery",
          id: "fourth-element-gallery",
          profile_id: "fourth-element-gallery-fourth-element-gallery-at-art-palm-springs-2019",
          partner_id: "5b294a72139b2147a3884c93",
        },
      ],
    },
    {
      letter: "G",
      exhibitors: [
        {
          name: "Gala Kvachnina",
          id: "gala-kvachnina",
          profile_id: "gala-kvachnina-gala-kvachnina-at-art-palm-springs-2019",
          partner_id: "5c61a2dc4a5c1c11b5e6b06d",
        },
        {
          name: "Gallery Harang",
          id: "gallery-harang",
          profile_id: "gallery-harang-gallery-harang-at-art-palm-springs-2019",
          partner_id: "5c2e50540cf1100007b437fd",
        },
        {
          name: "Gallery KAG",
          id: "gallery-kag-1",
          profile_id: "gallery-kag-1-gallery-kag-at-art-palm-springs-2019",
          partner_id: "5c2e50540cf1100007b43800",
        },
        {
          name: "Garis & Hahn",
          id: "garis-and-hahn",
          profile_id: "garis-and-hahn-garis-and-hahn-at-art-palm-springs-2019-1",
          partner_id: "51af7b4d275b249d7d000131",
        },
        {
          name: "George Billis Gallery",
          id: "george-billis-gallery",
          profile_id: "george-billis-gallery-george-billis-gallery-at-art-palm-springs-2019",
          partner_id: "59c2a37aab60ca1153bad4a2",
        },
      ],
    },
    {
      letter: "H",
      exhibitors: [
        {
          name: "HOHMANN",
          id: "hohmann",
          profile_id: "hohmann-hohmann-at-art-palm-springs-2019",
          partner_id: "54b40c11776f720e45900600",
        },
      ],
    },
    {
      letter: "I",
      exhibitors: [
        {
          name: "IAILA",
          id: "iaila",
          profile_id: "iaila-iaila-at-art-palm-springs-2019",
          partner_id: "5bc4d197e30f5c3b64638c27",
        },
      ],
    },
    {
      letter: "J",
      exhibitors: [
        {
          name: "Jane Kahan Gallery",
          id: "jane-kahan-gallery",
          profile_id: "jane-kahan-gallery-jane-kahan-gallery-at-art-palm-springs-2019",
          partner_id: "54b40c86776f720e45b50600",
        },
        {
          name: "Jody Klotz Fine Art",
          id: "jody-klotz-fine-art",
          profile_id: "jody-klotz-fine-art-jody-klotz-fine-art-at-art-palm-springs-2019",
          partner_id: "5bc65c788662557c2190f346",
        },
      ],
    },
    {
      letter: "K",
      exhibitors: [
        {
          name: "K + Y Gallery",
          id: "k-plus-y-gallery-1",
          profile_id: "k-plus-y-gallery-1-k-plus-y-gallery-at-art-palm-springs-2019",
          partner_id: "5698126e9acc8a3540000074",
        },
        {
          name: "Karyn Mannix Contemporary",
          id: "karyn-mannix-contemporary",
          profile_id: "karyn-mannix-contemporary-karyn-mannix-contemporary-at-art-palm-springs-2019",
          partner_id: "544fc8707261691b7b7a0000",
        },
        {
          name: "Katherina Rich Perlow Fine Arts",
          id: "katherina-rich-perlow-fine-arts",
          profile_id: "katherina-rich-perlow-fine-arts-katherina-rich-perlow-fine-arts-at-art-palm-springs-2019",
          partner_id: "5c2e50550cf1100007b43803",
        },
      ],
    },
    {
      letter: "L",
      exhibitors: [
        {
          name: "Lustre Contemporary",
          id: "lustre-contemporary",
          profile_id: "lustre-contemporary-lustre-contemporary-at-art-palm-springs-2019",
          partner_id: "5a57dbedcd530e517a560d9f",
        },
      ],
    },
    {
      letter: "M",
      exhibitors: [
        {
          name: "Maria Elena Kravetz",
          id: "maria-elena-kravetz",
          profile_id: "maria-elena-kravetz-maria-elena-kravetz-at-art-palm-springs-2019",
          partner_id: "54b40e2e776f720e45110700",
        },
      ],
    },
    {
      letter: "R",
      exhibitors: [
        {
          name: "Rebecca Hossack Art Gallery",
          id: "rebecca-hossack-art-gallery",
          profile_id: "rebecca-hossack-art-gallery-rebecca-hossack-art-gallery-at-art-palm-springs-2019",
          partner_id: "52276832ebad644079000263",
        },
      ],
    },
    {
      letter: "S",
      exhibitors: [
        {
          name: "Simard Bilodeau Contemporary",
          id: "simard-bilodeau-contemporary",
          profile_id: "simard-bilodeau-contemporary-simard-bilodeau-contemporary-at-art-palm-springs-2019",
          partner_id: "54cacdda7261695f96470200",
        },
      ],
    },
  ],
  __id: "RmFpcjphcnQtcGFsbS1zcHJpbmdzLTIwMTk=",
}
