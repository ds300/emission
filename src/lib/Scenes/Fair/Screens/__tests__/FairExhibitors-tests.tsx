import { MockRelayRenderer } from "lib/tests/MockRelayRenderer"
import { renderUntil } from "lib/tests/renderUntil"
import React from "react"
import { graphql } from "react-relay"
import { fairFixture } from "../../__fixtures__/fairExhibitors"
import { FairExhibitors } from "../FairExhibitors"

jest.unmock("react-relay")

it("renders properly", async () => {
  const tree = await renderUntil(
    wrapper => {
      return wrapper.find("FairExhibitors").length > 0
    },
    <MockRelayRenderer
      Component={FairExhibitors}
      query={graphql`
        query FairExhibitorsTestsQuery {
          fair(id: "art-basel-in-miami-beach-2018") {
            exhibitors_grouped_by_name {
              letter
              exhibitors {
                name
                id
                profile_id
              }
            }
          }
        }
      `}
      mockData={{
        fair: fairFixture,
      }}
    />
  )
  const htmlDom = tree.text()

  // Title
  expect(htmlDom).toContain("Exhibitors")

  // A "random" set of the exhibitors, we should show the letter
  // and all the exhibitors
  console.log(fairFixture.exhibitors_grouped_by_name)
  const exhibitors = fairFixture.exhibitors_grouped_by_name[1]
  expect(htmlDom).toContain(exhibitors.letter)

  // The fixtured data is too big for me to hand-edit all of the exhibitors
  // data into the shape we ask for
  const anyExhibitors = exhibitors.exhibitors as any[]

  // Because this uses flatlist you can't ask for all of the
  // exhibitors in the array
  expect(htmlDom).toContain(anyExhibitors[0].name)
  expect(htmlDom).toContain(anyExhibitors[1].name)
  expect(htmlDom).toContain(anyExhibitors[2].name)
})
