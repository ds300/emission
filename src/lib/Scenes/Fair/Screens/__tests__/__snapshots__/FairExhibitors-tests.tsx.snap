// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `"<rctscrollview sections=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" initialnumtorender=\\"10\\" maxtorenderperbatch=\\"10\\" scrolleventthrottle=\\"50\\" updatecellsbatchingperiod=\\"50\\" windowsize=\\"21\\" data=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" stickyheaderindices=\\"1,4\\"><view><view><view px=\\"2\\" mb=\\"2\\" pt=\\"85\\"><text font-family=\\"AGaramondPro-Regular\\" font-size=\\"32px\\" lineheight=\\"38px\\" class=\\"sc-bwzfXH uTxCW\\">Exhibitors</text></view></view><view><view px=\\"2\\" mb=\\"2\\"><text font-family=\\"AGaramondPro-Regular\\" font-size=\\"18px\\" lineheight=\\"26px\\" class=\\"sc-bwzfXH uTxCW\\">3</text></view></view><view><view px=\\"2\\"><view mb=\\"2\\"><view style=\\"opacity: 1;\\"><text font-family=\\"AGaramondPro-Regular\\" font-size=\\"16px\\" lineheight=\\"24px\\" class=\\"sc-bwzfXH uTxCW\\">Hello World</text></view></view></view></view><view></view><view><view px=\\"2\\" mb=\\"2\\"><text font-family=\\"AGaramondPro-Regular\\" font-size=\\"18px\\" lineheight=\\"26px\\" class=\\"sc-bwzfXH uTxCW\\">A</text></view></view><view><view px=\\"2\\"><view mb=\\"2\\"><view style=\\"opacity: 1;\\"><text font-family=\\"AGaramondPro-Regular\\" font-size=\\"16px\\" lineheight=\\"24px\\" class=\\"sc-bwzfXH uTxCW\\">Hello World</text></view></view></view></view><view><view px=\\"2\\"><view mb=\\"2\\"><view style=\\"opacity: 1;\\"><text font-family=\\"AGaramondPro-Regular\\" font-size=\\"16px\\" lineheight=\\"24px\\" class=\\"sc-bwzfXH uTxCW\\">Hello World</text></view></view></view></view><view><view px=\\"2\\"><view mb=\\"2\\"><view style=\\"opacity: 1;\\"><text font-family=\\"AGaramondPro-Regular\\" font-size=\\"16px\\" lineheight=\\"24px\\" class=\\"sc-bwzfXH uTxCW\\">Hello World</text></view></view></view></view><view><view px=\\"2\\"><view mb=\\"2\\"><view style=\\"opacity: 1;\\"><text font-family=\\"AGaramondPro-Regular\\" font-size=\\"16px\\" lineheight=\\"24px\\" class=\\"sc-bwzfXH uTxCW\\">Hello World</text></view></view></view></view><view><view px=\\"2\\"><view mb=\\"2\\"><view style=\\"opacity: 1;\\"><text font-family=\\"AGaramondPro-Regular\\" font-size=\\"16px\\" lineheight=\\"24px\\" class=\\"sc-bwzfXH uTxCW\\">Hello World</text></view></view></view></view><view><view px=\\"2\\"><view mb=\\"2\\"><view style=\\"opacity: 1;\\"><text font-family=\\"AGaramondPro-Regular\\" font-size=\\"16px\\" lineheight=\\"24px\\" class=\\"sc-bwzfXH uTxCW\\">Hello World</text></view></view></view></view><view style=\\"height: 0px;\\"></view></view></rctscrollview>"`;
